regression-tests/test-results/clang-12-c++20/pure2-bugfix-for-dependent-types-recursion.cpp.output

```diff
@@ -0,0 +1,13 @@
+pure2-bugfix-for-dependent-types-recursion.cpp2:2:13: error: unknown type name 'b'
+  using a = b;
+            ^
+pure2-bugfix-for-dependent-types-recursion.cpp2:3:13: error: unknown type name 'a'
+  using b = a;
+            ^
```

Comment:
Is it intended that these test cases fail?

- pure2-bugfix-for-dependent-types-recursion
- pure2-bugfix-for-dependent-types 

For some reason the latter works on GCC 14?

Reply:
`pure2-bugfix-for-dependent-types-recursion` should fail.
It tests that the algorithm detects recursio,
and returns `false` (i.e., "not a dependent type"),
to let the Cpp1 compiler give its diagnostic.

`pure2-bugfix-for-dependent-types` shouldn't fail on a modern compiler.
I see some are failing because they don't yet support optional `typename`.